src/App.tsx

```diff
@@ -1,26 +1,218 @@
+    try {
+      await deleteTodo(id);
+      setTodoList(prevTodoList => prevTodoList.filter(todo => todo.id !== id));
+      todoTitleRef.current?.focus();
+
+      return true;
+    } catch (error) {
+      setTodoError(ErrorType.UnableToDeleteTodo);
+
+      return false;
+    } finally {
+      setLoadingTodoIds(prev => prev.filter(todoId => todoId !== id));
+    }
+  };
+
+  const handleDeleteCompletedTodos = () => {
+    completedTodos.forEach(todo => handleDeleteTodo(todo.id));
+  };
+
+  const handleToggleAllTodos = () => {
+    if (uncompletedTodos.length > 0) {
+      uncompletedTodos.forEach(todo => {
+        handleUpdateTodo({ ...todo, completed: true });
+      });
+    } else {
+      todoList.forEach(todo => {
+        handleUpdateTodo({ ...todo, completed: false });
+      });
+    }
+  };
+
+  useEffect(() => {
+    const loadTodosFromServer = async () => {
+      try {
+        const todosFromServer = await getTodos();
+
+        setTodoList(todosFromServer);
+      } catch (error) {
+        setTodoError(ErrorType.UnableToLoadTodos);
+      }
+    };
+
+    loadTodosFromServer();
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">
-          React Todo App - Add and Delete
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <TodoHeader
+          todoTitleRef={todoTitleRef}
+          onAddTodo={handleAddTodo}
+          onToggleAllTodos={handleToggleAllTodos}
+          isLoading={isAddingLoading}
+          isAllTodoCompleted={isAllTodoCompleted}
+          isTodoListNotEmpty={!!todoList.length}
+        />
+
+        <section className="todoapp__main" data-cy="TodoList">
+          <TransitionGroup>
+            {filteredTodos.map(todo => (
+              <CSSTransition key={todo.id} timeout={300} classNames="item">
+                <TodoItem
+                  todo={todo}
+                  key={todo.id}
+                  onUpdateTodo={handleUpdateTodo}
+                  onDeleteTodo={handleDeleteTodo}
+                  isLoading={loadingTodoIds.includes(todo.id)}
+                />
+              </CSSTransition>
+            ))}
+
+            {tempTodo && (
+              <CSSTransition key={0} timeout={300} classNames="temp-item">
+                <TodoItem
+                  todo={tempTodo}
+                  onUpdateTodo={handleUpdateTodo}
+                  onDeleteTodo={handleDeleteTodo}
+                  isLoading={isAddingLoading}
+                />
+              </CSSTransition>
+            )}
+          </TransitionGroup>
+        </section>
```

Comment:
Consider will be better if you move this logic to the separate component, for example `TodoList`